code/modules/mob/living/silicon/ai/say.dm

```diff
@@ -58,106 +58,91 @@
+	if(!character_to_use)
+		return
+	var/max_characters = 200 // magic number but its the cap 15 allows
+	var/message = input(src, "Use the power of 15.ai to say anything! (200 character maximum)", "15.ai VOX System", src.last_announcement) as text|null
 
 	if(!message || announcing_vox > world.time)
 		return
 
-	last_announcement = message
-
 	if(incapacitated())
 		return
 
 	if(control_disabled)
 		to_chat(src, "<span class='warning'>Wireless interface disabled, unable to interact with announcement PA.</span>")
 		return
 
-	var/list/words = splittext(trim(message), " ")
-	var/list/incorrect_words = list()
-
-	if(words.len > 30)
-		words.len = 30
-
-	for(var/word in words)
-		word = lowertext(trim(word))
-		if(!word)
-			words -= word
-			continue
-		if(!GLOB.vox_sounds[word])
-			incorrect_words += word
-
-	if(incorrect_words.len)
-		to_chat(src, "<span class='notice'>These words are not available on the announcement system: [english_list(incorrect_words)].</span>")
+	if(length(message) > max_characters)
+		to_chat(src, "<span class='notice'>You have too many characters! You used [length(message)] characters, you need to lower this to [max_characters] or lower.</span>")
+		return
+	var/regex/check_for_bad_chars = regex("\[^a-zA-Z!?.,' :\]+")
+	if(check_for_bad_chars.Find(message))
+		to_chat(src, "<span class='notice'>These characters are not available on the 15.ai system: [english_list(check_for_bad_chars.group)].</span>")
 		return
 
-	announcing_vox = world.time + VOX_DELAY
-
-	log_game("[key_name(src)] made a vocal announcement with the following message: [message].")
-
-	for(var/word in words)
-		play_vox_word(word, src.z, null)
-
-
-/proc/play_vox_word(word, z_level, mob/only_listener)
-
-	word = lowertext(word)
+	var/emotion_to_use = "Neutral"
+	switch(character_to_use)
+		if("GLaDOS (Homicidal)")
+			emotion_to_use = "Homicidal"
+			character_to_use = "GLaDOS"
+		if("GLaDOS (Neutral)")
+			character_to_use = "GLaDOS"
+		if("Twilight Sparkle (Happy)")
+			emotion_to_use = "Happy"
+			character_to_use = "Twilight Sparkle"
+		if("Twilight Sparkle (Neutral)")
+			character_to_use = "Twilight Sparkle"
+	last_announcement = message
 
-	if(GLOB.vox_sounds[word])
+	announcing_vox = world.time + VOX_DELAY
 
-		var/sound_file = GLOB.vox_sounds[word]
-		var/sound/voice = sound(sound_file, wait = 1, channel = CHANNEL_VOX)
+	log_game("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	message_admins("[key_name(src)] started making a 15.AI announcement with the following message: [message]")
+	play_vox_word(message, character_to_use, emotion_to_use, src, src.z, null)
+
+
+/proc/play_vox_word(message, character, emotion, mob/living/speaker, z_level, mob/only_listener)
+	var/api_url = "https://api.fifteen.ai/app/getAudioFile"
+
+	var/datum/http_request/req = new()
+	var/vox_audio_number = rand(1,9999)

```

Comment:
This is ass. You want each number to only be used once.
Even something like this would be way better:
```
var/static/next_in_line = 0
var/vox_audio_number = next_in_line++
```